src/components/PostsList.tsx

```diff
@@ -1,56 +1,58 @@
 import classNames from 'classnames';
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { set as setSelectedPost } from '../features/selectedPost/selectedPost';
 import { Post } from '../types/Post';
 
-type Props = {
-  posts: Post[],
-  selectedPostId?: number,
-  onPostSelected: (post: Post | null) => void,
-};
+export const PostsList: React.FC = () => {
+  const { posts } = useAppSelector(state => state.posts);
+  const dispatch = useAppDispatch();
```

Comment:
Destructure the state to get the `posts` and `selectedPost` properties in a single `useAppSelector` call.